packages/smooth_app/lib/pages/preferences/user_preferences_languages_list.dart

```
@@ -0,0 +1,430 @@
+        LanguageName(englishName: 'Wolof', nameInLanguage: 'Wolof'),
+    OpenFoodFactsLanguage.XHOSA:
+        LanguageName(englishName: 'Xhosa', nameInLanguage: 'isiXhosa'),
+    OpenFoodFactsLanguage.YIDDISH:
+        LanguageName(englishName: 'Yiddish', nameInLanguage: 'יידיש'),
+    OpenFoodFactsLanguage.YORUBA:
+        LanguageName(englishName: 'Yoruba', nameInLanguage: 'Yoruba'),
+    OpenFoodFactsLanguage.CHINESE:
+        LanguageName(englishName: 'Chinese', nameInLanguage: '中文'),
+    OpenFoodFactsLanguage.ZHUANG_LANGUAGES:
+        LanguageName(englishName: 'Zhuang', nameInLanguage: 'Zhuang'),
+    OpenFoodFactsLanguage.ZULU:
+        LanguageName(englishName: 'Zulu', nameInLanguage: 'ខ្មែរ'),
+  };
+
+  List<Pair<String, OpenFoodFactsLanguage>> getSupportedLanguagesEnglishName() {
+    final List<Pair<String, OpenFoodFactsLanguage>> languages =
+        <Pair<String, OpenFoodFactsLanguage>>[];
+    openFoodFactsLanguagesList.forEach(
+        (OpenFoodFactsLanguage language, LanguageName languageName) => <void>{
+              if (delegate.isSupported(Locale(language.code)))
+                <void>{
+                  languages.add(Pair<String, OpenFoodFactsLanguage>(
+                      first: languageName.englishName, second: language))
+                }
+            });
+    return languages;
+  }
+
+  LanguageName getLanguageNameFromLangCode(String langCode) {
+    OpenFoodFactsLanguage openFoodFactsLanguage = OpenFoodFactsLanguage.ENGLISH;
+
+    openFoodFactsLanguagesList.forEach(
+        (OpenFoodFactsLanguage language, LanguageName languageName) => <void>{
+              if (language.code == langCode) openFoodFactsLanguage = language
+            });
+    return openFoodFactsLanguagesList[openFoodFactsLanguage]!;
+  }
+
+  LanguageName getLanguageName(OpenFoodFactsLanguage offlc) {
+    if (openFoodFactsLanguagesList.containsKey(offlc)) {
+      return openFoodFactsLanguagesList[offlc]!;
+    }
+    // Unreachable Code
```

If it's unreachable please throw an exception or log an error with sentry (something like `Log.e(`)

---

Thanks for the suggestion.